### PR TITLE
Use metadata to denote parameters

### PR DESCRIPTION
**Pull request by shashi**
fixes https://github.com/SciML/ModelingToolkit.jl/issues/881

### CODE REVIEW

**Comment by shashi** [comment]
requires https://github.com/JuliaSymbolics/Symbolics.jl/pull/163

File: src/parameters.jl [hunk 1]
```diff
@@ -1,24 +1,35 @@
-import SymbolicUtils: symtype, term
-struct Parameter{T} end
+import SymbolicUtils: symtype, term, hasmetadata
+struct MTKParameterCtx end
 
+isparameter(x::Num) = isparameter(value(x))
+isparameter(x::Symbolic) = hasmetadata(x, MTKParameterCtx) && getmetadata(x, MTKParameterCtx)
```
Comment (YingboMa) [inline src/parameters.jl:5]: Can we do
```suggestion
isparameter(x::Symbolic) = getmetadata(x, MTKParameterCtx, false)
```
?

**Review by YingboMa — COMMENTED** [review]
Could you update the relavent docs as well?

File: src/systems/abstractsystem.jl [hunk 2]
```diff
@@ -274,7 +274,7 @@ function renamespace(namespace, x)
     elseif istree(x)
         renamespace(namespace, operation(x))(arguments(x)...)
     elseif x isa Sym
-        Sym{symtype(x)}(renamespace(namespace,nameof(x)))
+        @set! x.name = renamespace(namespace,nameof(x))
```
Comment (shashi) [inline src/systems/abstractsystem.jl:277]: The previous behavior was to actually not propagate the metadata... This is the cause of the current test failure I believe.

**Comment by shashi** [comment]
@YingboMa legit ci fail

**Pull request merged**